dali/operators/math/expressions/expression_impl_factory.h

```diff
@@ -38,9 +38,83 @@ namespace dali {
 
 #define ALLOWED_BIN_OPS                                                                            \
   (ArithmeticOp::add, ArithmeticOp::sub, ArithmeticOp::mul, ArithmeticOp::div, ArithmeticOp::fdiv, \
-      ArithmeticOp::mod, ArithmeticOp::eq, ArithmeticOp::neq, ArithmeticOp::lt, ArithmeticOp::leq, \
-      ArithmeticOp::gt, ArithmeticOp::geq, ArithmeticOp::bit_and, ArithmeticOp::bit_or,            \
-      ArithmeticOp::bit_xor)
+  ArithmeticOp::mod, ArithmeticOp::min, ArithmeticOp::max, ArithmeticOp::eq, ArithmeticOp::neq,    \
+  ArithmeticOp::lt, ArithmeticOp::leq, ArithmeticOp::gt, ArithmeticOp::geq,                        \
+  ArithmeticOp::bit_and, ArithmeticOp::bit_or, ArithmeticOp::bit_xor)
+
+#define ALLOWED_TERNARY_OPS \
+  (ArithmeticOp::clamp)
+
+namespace expression_detail {
+
+/**
+ * @brief Pass through as a pointer to T or return the pointed value based on `as_ptr`
+ */
+template <bool as_ptr, typename T>
+DALI_HOST_DEV std::enable_if_t<as_ptr, const T*> Pass(const T* ptr) {
```

Comment:
Maybe this `Pass` can be moved to a common utility.

Reply:
I'm not sure how useful such utility would be, I's a bit of implementation detail here, but the name is rather generic. I didn't want to pollute the dali namespace.